File: cases/taint/go/alias-propagation-negative/alias_propagation_negative.go
package dataflowbench

type Holder struct {
	Value int
}

func dfb_source() int { // DFB-SOURCE: alias-propagation-input
	return 1
}

func dfb_sink(value int) {} // DFB-SINK: alias-propagation-sink

func run() {
	original := Holder{}
	alias := &original // DFB-WITNESS: alias-propagation-alias
	distinct := Holder{}
	distinct.Value = 0
	original.Value = dfb_source() // DFB-WITNESS: alias-propagation-store
	_ = alias
	dfb_sink(distinct.Value)
}

File: cases/taint/php/alias-propagation-negative/alias_propagation_negative.php
<?php
class Holder
{
    public string $value = "clean";
}

function dfb_source(): string { // DFB-SOURCE: alias-propagation-input
    return "tainted";
}

function dfb_sink(string $value): void {} // DFB-SINK: alias-propagation-sink

function run(): void {
    $original = new Holder();
    $alias = $original; // DFB-WITNESS: alias-propagation-alias
    $distinct = new Holder();
    $original->value = dfb_source(); // DFB-WITNESS: alias-propagation-store
    dfb_sink($distinct->value);
}
